SDWebImage/UIImage+WebP.m

```diff
@@ -24,6 +21,82 @@ static void FreeImageData(void *info, const void *data, size_t size)
 @implementation UIImage (WebP)
 
 + (UIImage *)sd_imageWithWebPData:(NSData *)data {
+    
+    UIImage *image = [UIImage sd_animatedWebPImageWithWebPData:data];
+    if (image) {
+        return image;
+    } else {
+        return [UIImage sd_webpImageWithWithWebPData:data];
+    }
+    
+}
+
++ (UIImage *)sd_animatedWebPImageWithWebPData:(NSData *)data {
+    UIImage *result = nil;
```

Comment:
Implementation of sd_animatedWebPImageWithWebPData is inconsistent with sd_webpImageWithWithWebPData. See handling of checking
`if (!WebPInitDecoderConfig(&config)) {` in sd_webpImageWithWithWebPData v.s. just calling `WebPInitDecoderConfig(&config);`in sd_animatedWebPImageWithWebPData.

Also you should remove redundant code by checking the ANIMATION_FLAG only in one location in sd_imageWithWebPData. You could forward WebPData directly to both methods.